### PR TITLE
Error on untyped enum values in Postgres

### LINKED ISSUE #4046: Error on untyped enum values in Postgres
### Describe your issue

I'm actually not sure if I have to post the bug here or to npgsql, but the issue is quite severe for us, so, if you think it's not possible to make anything with that issue on the side of linq2db, please, let me know in order to recreate the issues on the npgsql side.

Suppose we have a database with the table:

```sql
create table person
(
    name     text    not null,
    category integer not null
);
```

and the corresponding generated class: 
```c#
[Table("person", Schema = "public")]
	public partial class Person
	{
		[Column("name"    , CanBeNull = false)] public string Name     { get; set; } = null!; // text
		[Column("category"                   )] public int    Category { get; set; } // integer
	}
```

Let the category be logically the enum but not having the mapping in the database to the custom enum type (it's quite common case).

In the described case it's convenient to convert the category to C# enum:
```c#
  public enum PersonCategory
    {
        Friends = 1,
        Relatives = 2
    }
```
like 
```c#
db.People.Select(p => new { p.Name, Category = (PersonCategory)p.Category })
```

And after that, let's suppose we have some third-party lib that applies filters to our IQueryables (exactly our production case). Usually, such libs produce the code with untyped enums. Let's model that situation in the following test:
```c#
[Test]
        public async Task FilterWithEnumAsObjectTest_Fail()
        {
            var db = new EntitiesDataConnection();

           //The enum is untyped
            object categoryParam = PersonCategory.Friends;

            var result = await db.People
                .Select(p => new { p.Name, Category = (PersonCategory)p.Category })
                .Where(p => p.Category.Equals(categoryParam))
                .CountAsync();
        }
```
That test will fail with the exception:
```
System.ArgumentException : A PostgreSQL type with the name 'person_category' was not found in the database
   at Npgsql.Internal.NpgsqlDatabaseInfo.GetPostgresTypeByName(String pgName)
   at Npgsql.TypeMapping.TypeMapper.ResolveByClrType(Type type)
   at Npgsql.NpgsqlParameter.ResolveHandler(TypeMapper typeMapper)
   at Npgsql.NpgsqlParameterCollection.ProcessParameters(TypeMapper typeMapper, Boolean validateValues, CommandType commandType)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteReader(CommandBehavior behavior, Boolean async, CancellationToken cancellationToken)
   at Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.ExecuteReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.ExecuteDataReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.ExecuteDataReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReaderAsync(CancellationToken cancellationToken)
   at LinqToDB.Linq.QueryRunner.ExecuteElementAsync[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, CancellationToken cancellationToken)
   at LinqToDB.Linq.QueryRunner.ExecuteElementAsync[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, CancellationToken cancellationToken)
   at LinqToDB.Linq.ExpressionQuery`1.LinqToDB.Async.IQueryProviderAsync.ExecuteAsync[TResult](Expression expression, CancellationToken cancellationToken)
   at GtG.Core.Tests.Test.FilterWithEnumAsObjectTest_Fail() in C:\Sources\B2B\bgs-platform\BigGameSolutions.Tests\BetWinRefundGenericTests.cs:line 49
   at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.GetResult()
   at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
   at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
   at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
   at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0()
   at NUnit.Framework.Internal.Commands.DelegatingTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)
```
And the generated SQL script is obviously not correct (AFAIK there's just no syntax allowing to define the type "Enum" of the variable): 
```
BeforeExecute
-- GtG.Core.Db PostgreSQL.9.5 PostgreSQL (asynchronously)
DECLARE @categoryParam Enum -- Object
SET     @categoryParam = 1

SELECT
	Count(*)
FROM
	"public".person p
WHERE
	p.category = :categoryParam
```

Of course, there's a "kind of" a workaround - just to use the strongly typed enums, but, as I told before, that's not the case when the code is applied by some third-party library which intentionally uses untyped objects in order to cover all possible cases.
```c#
         [Test]
        public async Task FilterWithEnumStronglyTypedTest_Success()
        {
            var db = new EntitiesDataConnection();

            PersonCategory categoryParam = PersonCategory.Friends;

            var result = await db.People
                .Select(p => new { p.Name, Category = (PersonCategory)p.Category })
                .Where(p => p.Category.Equals(categoryParam))
                .CountAsync();
        }
```

`Linq To DB` version: *4.x, 5.x*

Database (with version): *Postgres 14.x*

ADO.NET Provider (with version): *Npgsql 7.x* 

Is it possible to fix that somehow on the side of linq2db?

### PR DESCRIPTION
**Pull request by igor-tkachev**
It does not fix #4046. We are not going to change the default behavior as this is too general. However, we add an option for a workaround.

```c#
MappingSchema.Default.SetConverter<object?,DataParameter>(
    o => new(null, o is Enum e ? MappingSchema.Default.EnumToValue(e) : o, DataType.Undefined));
```

### CODE REVIEW

File: Build/Azure/scripts/clickhouse.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-docker run -d --name clickhouse --ulimit nofile=262144:262144 -p 8123:8123 -p 9000:9000 -p 9004:9004 -p 9005:9005 -e CLICKHOUSE_USER=testuser -e CLICKHOUSE_PASSWORD=testuser clickhouse/clickhouse-server:latest
+docker run -d --name clickhouse --ulimit nofile=262144:262144 -p 8123:8123 -p 9000:9000 -p 9004:9004 -p 9005:9005 -e CLICKHOUSE_USER=testuser -e CLICKHOUSE_PASSWORD=testuser clickhouse/clickhouse-server:23.1.3.5
```
Comment (MaceWindu) [inline Build/Azure/scripts/clickhouse.sh:3]: revert

File: Build/Azure/scripts/clickhouse.sh [hunk 2]
```diff
@@ -31,3 +31,5 @@ docker exec clickhouse clickhouse-client --multiquery --host 127.0.0.1 -u testus
 #docker exec clickhouse clickhouse-client --multiquery --host 127.0.0.1 -u testuser --password testuser -q "CREATE DATABASE testdb4"
 
 docker logs clickhouse
+
+docker exec clickhouse clickhouse-client --multiquery --host 127.0.0.1 -u testuser --password testuser -q "SELECT version()"
```
Comment (MaceWindu) [inline Build/Azure/scripts/clickhouse.sh:35]: remove

**Review by MaceWindu — APPROVED** [review]
revert clickhouse setup changes

**Review by viceroypenguin — APPROVED** [review]

**Review by MaceWindu — APPROVED** [review]

**Pull request merged**